marshal.go

```diff
@@ -0,0 +1,103 @@
+package putty
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/sha1"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+)
+
+func marshal(val interface{}) (data []byte, keySize int, err error) {
+	v := reflect.ValueOf(val).Elem()
+	buf := bytes.NewBuffer([]byte{})
+
+	err = writeField(v, buf)
+	data = buf.Bytes()
+	keySize = buf.Len()
+	return
+}
+
+func addPadding(data []byte) []byte {
```

Comment:
This is how putty is written... I thought that same thing, but realized that this was added to only to prevent the CBC bits from being exposed and cannot be useful for reversing the sha1 or the CBC without first having the key bytes

Reply:
The PR is correct, no change needed.